solr/core/src/java/org/apache/solr/handler/component/ResponseBuilder.java

```diff
@@ -60,7 +60,6 @@ public class ResponseBuilder {
   public boolean doExpand;
   public boolean doStats;
   public boolean doTerms;
-  public boolean doAnalytics;
```

Comment:
observation only and I am not aware of such a use case: if someone wanted to take the removed analytics module code and have it locally or as a package plugin or so then the removal of the `ResponseBuilder` members and setter/getters here would present a small challenge. aside from patching the equivalent of what is removed here into a local variant of the `ResponseBuilder`, it's possible to have a custom response builder and the notes on the https://issues.apache.org/jira/browse/SOLR-13511 illustrate how a custom response builder/custom search handler combination can be done and configured.

Reply:
Thanks for commenting on this. I was surprised to find analytics specific vars in ResponseBuilder, and could not find any other use of these, so I removed them. If someone were to package the component as a package and maintain it on GitHub, it would be nice if this was more generic.

From SOLR-13511 I understand that you can provide your own custom ResponseBuilder if you own the SearchHandler. Fair enough, could be part of that plugin package.

Or would it be possible to have some kind of `Map<Sting,Object> responseContext` structure on ResponseBuilder where plugins could register arbitrary data? I'm on thin ice here...

Reply:
Yeah unfortunately this was necessary at the time, although it is very regrettable.


> From [SOLR-13511](https://issues.apache.org/jira/browse/SOLR-13511) I understand that you can provide your own custom ResponseBuilder if you own the SearchHandler. Fair enough, could be part of that plugin package.

Yeah, this seems like a pretty easy win. But a `Map<Sting,Object> responseContext` would also work pretty easily. Basically there just needs to be a way for SearchComponents to store customState in the response builder.